### PR TITLE
perlPackages.AppSqitch: 1.4.0 -> 1.4.1

### PR DESCRIPTION
**Pull request by akegalj**
Bump sqitch tool from v1.4.0 to v1.4.1

This realease https://github.com/sqitchers/sqitch/releases/tag/v1.4.1 mostly contains bug fixes. Most notably:

> Updated the locale configuration to fix issues in more recent versions of Perl, and added tests to ensure that the sqitch CLI executes and properly emits localized messages (except on Windows, where the language codes are incompatible).

which is related to sqitch v1.4.0 failing on some machines due too bug in setting a locale. This was fixed here https://github.com/sqitchers/sqitch/issues/806#issuecomment-1908340792 and pushed to v1.4.1.

Other comments from release notes don't look like a breaking change. I have checked note:

> Tests now require Test::Warn 0.31 or later, as newline handling issues cause test failures in earlier versions. Thanks to Slaven Rezić for the test reports and for identifying the issue.

And this doesn't seem like a breaking change as `TestWarn` is at version `0.37` https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/perl-packages.nix#L26217

Note: This package doesn't have tests so no automated tests are run

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by m1dugh — APPROVED** [review]

**Comment by zakame** [comment]
@ofborg build perl540Packages.AppSqitch

**Comment by zakame** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 351171`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>perl538Packages.AppSqitch</li>
    <li>perl538Packages.AppSqitch.devdoc</li>
    <li>perl540Packages.AppSqitch</li>
    <li>perl540Packages.AppSqitch.devdoc</li>
    <li>sqitchMysql</li>
    <li>sqitchPg</li>
  </ul>
</details>

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -992,11 +992,11 @@ with self; {
...
+    version = "1.4.1";
     pname = "App-Sqitch";
     src = fetchurl {
-      url = "mirror://cpan/authors/id/D/DW/DWHEELER/App-Sqitch-v1.4.0.tar.gz";
-      hash = "sha256-sNs4cDH3dWJmLgA7xV16EComOAtK1/25qKO61XaeUBw=";
+      url = "mirror://cpan/authors/id/D/DW/DWHEELER/App-Sqitch-v1.4.1.tar.gz";
+      hash = "sha256-yvMcyPdy46TJ1LP/Oo9oSm61sbPCYfTdwPkKiMNgB8Y=";
     };
     buildInputs = [ CaptureTiny TestDeep TestDir TestException TestFile TestFileContents TestMockModule TestMockObject TestNoWarnings TestWarn ];
     propagatedBuildInputs = [ Clone ConfigGitLike DBI DateTime EncodeLocale HashMerge IOPager IPCRun3 IPCSystemSimple ListMoreUtils PathClass PerlIOutf8_strict PodParser StringFormatter StringShellQuote TemplateTiny Throwable TypeTiny URIdb libintl-perl ];
```
[the remark this answers is not in the document]
Comment (stigtsp) [reply pkgs/top-level/perl-packages.nix:1002]: I think we should enable tests here to pick up on these kinds of errors, this patch sorts that:

```diff
diff --git a/pkgs/top-level/perl-packages.nix b/pkgs/top-level/perl-packages.nix
index 37a5261383f3..a007e54fab50 100644
--- a/pkgs/top-level/perl-packages.nix
+++ b/pkgs/top-level/perl-packages.nix
@@ -998,9 +998,8 @@ with self; {
       url = "mirror://cpan/authors/id/D/DW/DWHEELER/App-Sqitch-v1.4.1.tar.gz";
       hash = "sha256-yvMcyPdy46TJ1LP/Oo9oSm61sbPCYfTdwPkKiMNgB8Y=";
     };
-    buildInputs = [ CaptureTiny TestDeep TestDir TestException TestFile TestFileContents TestMockModule TestMockObject TestNoWarnings TestWarn ];
-    propagatedBuildInputs = [ Clone ConfigGitLike DBI DateTime EncodeLocale HashMerge IOPager IPCRun3 IPCSystemSimple ListMoreUtils PathClass PerlIOutf8_strict PodParser StringFormatter StringShellQuote TemplateTiny Throwable TypeTiny URIdb libintl-perl ];
-    doCheck = false;  # Can't find home directory.
+    buildInputs = [ CaptureTiny TestExit TestDeep TestDir TestException TestFile TestFileContents TestMockModule TestMockObject TestNoWarnings TestWarn ];
+    propagatedBuildInputs = [ Clone ConfigGitLike DBI DateTime EncodeLocale HashMerge IOPager IPCRun3 IPCSystemSimple ListMoreUtils PathClass PerlIOutf8_strict PodParser StringFormatter StringShellQuote TemplateTiny Throwable TypeTiny URIdb libintl-perl AlgorithmBackoff ];
     meta = {
       description = "Sensible database change management";
       homepage = "https://sqitch.org";
@@ -22105,6 +22104,21 @@ with self; {
     };
   };

+  ReturnMultiLevel = buildPerlPackage {
+    pname = "Return-MultiLevel";
+    version = "0.08";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/P/PL/PLICEASE/Return-MultiLevel-0.08.tar.gz";
+      hash = "sha256-UbGu8wxcQAn2QCZ6CFiSEuh9zRAYAPDSD5xjXJ/+iKE=";
+    };
+    buildInputs = [ TestFatal ];
+    meta = {
+      homepage = "https://metacpan.org/pod/Return::MultiLevel";
+      description = "Return across multiple call levels";
+      license = with lib.licenses; [ artistic1 gpl1Plus ];
+    };
+  };
+
   ReturnValue = buildPerlPackage {
     pname = "Return-Value";
     version = "1.666005";
@@ -25039,6 +25053,20 @@ with self; {
     };
   };

+  TestExit = buildPerlPackage {
+    pname = "Test-Exit";
+    version = "0.11";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/A/AR/ARODLAND/Test-Exit-0.11.tar.gz";
+      hash = "sha256-+9qS034EgdGO68geSNAlIotXGExZstWm9r34cELox7I=";
+    };
+    propagatedBuildInputs = [ ReturnMultiLevel ];
+    meta = {
+      description = "Test whether code exits without terminating testing";
+      license = with lib.licenses; [ artistic1 gpl1Plus ];
+    };
+  };
+
   TestExpect = buildPerlPackage {
     pname = "Test-Expect";
     version = "0.34";
```

[the remark this answers is not in the document]
Comment (akegalj) [reply pkgs/top-level/perl-packages.nix:1002]: > Apparently it also needs a dep on `AlgorithmBackoff`:
> 
> ```
> Running phase: buildPhase
> Checking prerequisites...
>   requires:
>     !  Algorithm::Backoff::Exponential is not installed
> ```
> 
> (There's also a`test_requires` on Test-Exit, but doesn't seem to break anything in this case.)

How did you get that info? From CI log? (trying to understand for future PRs)

[the remark this answers is not in the document]
Comment (akegalj) [reply pkgs/top-level/perl-packages.nix]: @stigtsp I have pushed your patch to this PR

[the remark this answers is not in the document]
Comment (akegalj) [reply pkgs/top-level/perl-packages.nix]: @zakame 
> nixpkgs-review pr 351171

thanks for the command. I was using `nixpkgs-review rev HEAD` which seems to do something different (so I aborted that command when I tried it yesterday).

[the remark this answers is not in the document]
Comment (akegalj) [reply pkgs/top-level/perl-packages.nix]: With that patch "required" prerequisites are gone. There are now "recommends":
```
patching ./t/firebird.t...
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Checking prerequisites...
  recommends:
    *  Class::XSAccessor is not installed
    *  Template is not installed
    *  Type::Tiny::XS is not installed

```
but my assumption is that these don't have to be added. Not sure what that even means, dependency to be recommended :|

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -22105,6 +22104,21 @@ with self; {
...
+      hash = "sha256-UbGu8wxcQAn2QCZ6CFiSEuh9zRAYAPDSD5xjXJ/+iKE=";
+    };
+    buildInputs = [ TestFatal ];
+    meta = {
+      homepage = "https://metacpan.org/pod/Return::MultiLevel";
+      description = "Return across multiple call levels";
+      license = with lib.licenses; [ artistic1 gpl1Plus ];
+    };
+  };
+
```
Comment (zakame) [inline pkgs/top-level/perl-packages.nix:22121]: Please commit separately under `perlPackage.ReturnMultiLevel: init at 0.08`

File: pkgs/top-level/perl-packages.nix [hunk 3]
```diff
@@ -25039,6 +25053,20 @@ with self; {
...
+      url = "mirror://cpan/authors/id/A/AR/ARODLAND/Test-Exit-0.11.tar.gz";
+      hash = "sha256-+9qS034EgdGO68geSNAlIotXGExZstWm9r34cELox7I=";
+    };
+    propagatedBuildInputs = [ ReturnMultiLevel ];
+    meta = {
+      description = "Test whether code exits without terminating testing";
+      license = with lib.licenses; [ artistic1 gpl1Plus ];
+    };
+  };
+
```
Comment (zakame) [inline pkgs/top-level/perl-packages.nix:25069]: Also commit this separately as `perlPackages.TestExit: init at 0.11`

**Review by zakame — CHANGES_REQUESTED** [review]

File: pkgs/top-level/perl-packages.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (zakame) [reply pkgs/top-level/perl-packages.nix]: Yep that should be fine - we can leave that to the consumers of the package to add recommended deps via overlays.

File: pkgs/top-level/perl-packages.nix (hunk 3 above)
Comment (akegalj) [reply pkgs/top-level/perl-packages.nix:25069]: I have rearanged the commits in git so they appear in a correct order based on dependencies:
```
[akegalj@x220:~/projects/nixpkgs]$ git log --pretty=format:%s HEAD^~2..HEAD
perlPackages.AppSqitch: 1.4.0 -> 1.4.1
perlPackages.TestExit: init at 0.11
perlPackages.ReturnMultiLevel: init at 0.08
```

**Review by stigtsp — APPROVED** [review]
Looks good to me, thanks for the contribution and welcome to the project :)

**Review by zakame — APPROVED** [review]
LGTM and welcome! 🎉 🚀

**Pull request merged**